osutil/disks/disks_linux.go

```diff
@@ -144,11 +144,24 @@ func diskFromUdevProps(deviceIdentifier string, deviceIDType string, props map[s
 	// create the full path by pre-pending /sys, since udev doesn't include /sys
 	devpath = filepath.Join(dirs.SysfsDir, devpath)
 
+	// TODO: this doesn't seem to work for /dev/mapper/loop1p1 devices like we
```

Comment:
did you mean /dev/loop1p1 maybe, I've never seen the loopback partitions nodes created under /dev/mapper

Reply:
hmm, it seems actually looking at the spread test I was wrong, I assumed the spread test was using kpartx but it seems that the spread test uses losetup directly, when done with losetup directly the way the spread test does it, the partition nodes are indeed created normally, but when I do `sudo kpartx -av foo.img`, then there are not partition nodes created at `/dev/loop235p1` the way they are with the spread test.

```
$ sudo kpartx -av foo.img
add map loop235p1 (253:0): 0 8 linear 7:235 2048
add map loop235p2 (253:1): 0 2097152 linear 7:235 2056
$ ls -lah /dev/loop235*
brw-rw---- 1 root disk 7, 235 Oct 11 10:47 /dev/loop235
$ ls -lah /dev/mapper/loop235p*
lrwxrwxrwx 1 root root 7 Oct 11 10:47 /dev/mapper/loop235p1 -> ../dm-0
lrwxrwxrwx 1 root root 7 Oct 11 10:47 /dev/mapper/loop235p2 -> ../dm-1
$ ls -lah /sys/$(udevadm info --query property /dev/loop235 | grep -Po 'DEVPATH=\K.*')/$(basename $(udevadm info --query property /dev/loop235 | grep -Po 'DEVNAME=\K.*'))*
ls: cannot access '/sys//devices/virtual/block/loop235/loop235*': No such file or directory
```

so I think the answer here is that I need to stop using kpartx with my scripts and should switch to using losetup the same way that the spread test does